### PR TITLE
Render elements in module doc and sanity test for sub-options

### PR DESCRIPTION
**Pull request by ganeshrn**
##### SUMMARY

*  Add support to render module elements value in ansible-doc output and
   module HTML doc
*  Add validate-module sanity test for suboptions.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

docs/docsite/rst/dev_guide/testing_validate-modules.rst 
docs/templates/plugin.rst.j2
test/sanity/validate-modules/main.py
test/sanity/validate-modules/schema.py

##### ADDITIONAL INFORMATION

The element's value in doc is displayed as
elements_out

### CODE REVIEW

**Comment by ganeshrn** [comment]
The validate module sanity failures in shippable are valid failures since this PR now validates recursively over sub-options. Will add it to the ignore list modules for now

**Review by NilashishC — APPROVED** [review]

**Comment by webknjaz** [comment]
Hey @ganeshrn, this PR is out of sync with devel. Could you please fix this?

File: test/sanity/validate-modules/ignore.txt [hunk 1]
```diff
@@ -1,14 +1,15 @@
-lib/ansible/modules/cloud/alicloud/ali_instance_info.py E337
-lib/ansible/modules/cloud/alicloud/ali_instance_info.py E338
 lib/ansible/modules/cloud/alicloud/ali_instance.py E337
 lib/ansible/modules/cloud/alicloud/ali_instance.py E338
+lib/ansible/modules/cloud/alicloud/ali_instance_info.py E337
```
Comment (webknjaz) [inline test/sanity/validate-modules/ignore.txt]: Why reorder this?

**Comment by webknjaz** [comment]
Could you please move out the ignore entries reordering into a separate PR?

File: test/sanity/validate-modules/main.py [hunk 2]
```diff
@@ -1144,18 +1144,22 @@ def _validate_ansible_module_call(self, docs):
...
 
-        try:
-            add_fragments(docs, self.object_path, fragment_loader=fragment_loader)
-        except Exception:
-            # Cannot merge fragments
-            return
+        if context is None:
+            context = []
+
+        if not context:
```
Comment (webknjaz) [inline test/sanity/validate-modules/main.py]: Plz move this check inside the try block to reduce nesting.

**Comment by samccann** [comment]
@ganeshrn can you point me to some modules in your branch that will exercise this new docs code so I can take a look?

**Comment by ganeshrn** [comment]
@webknjaz Ignore list file is moved out of validate-modules folder as part of https://github.com/ansible/ansible/pull/59618

**Comment by ganeshrn** [comment]
@webknjaz I have fixed your review comments. 

@webknjaz @samccann 
Since this PR has a considerable amount of changes in ignore-changes.txt  I had to rebase it multiple times due to merge conflicts. 
Since the CI is passing now, I will go ahead and merge this PR which will unblock merge for a number of other network resource module PR's. 

If there are any more review comments/issues I will fix it is a separate PR.

**Pull request merged**